plugins/wasmedge_zlib/zlibfunc.cpp

```diff
@@ -0,0 +1,1411 @@
+  });
+
+  if (ZRes != Z_OK)
+    Env.ZStreamMap.erase(It);
+
+  return ZRes;
+}
+
+Expect<int32_t>
+WasmEdgeZlibDeflateReset::body(const Runtime::CallingFrame &Frame,
+                               uint32_t ZStreamPtr) {
+
+  const auto ZRes = SyncRun(Env, ZStreamPtr, Frame, [&](z_stream *HostZStream) {
+    return deflateReset(HostZStream);
+  });
+
+  return ZRes;
+}
+
+Expect<int32_t>
+WasmEdgeZlibDeflateParams::body(const Runtime::CallingFrame &Frame,
+                                uint32_t ZStreamPtr, int32_t Level,
+                                int32_t Strategy) {
+
+  const auto ZRes = SyncRun(Env, ZStreamPtr, Frame, [&](z_stream *HostZStream) {
+    return deflateParams(HostZStream, Level, Strategy);
+  });
+
+  return ZRes;
+}
+
+Expect<int32_t> WasmEdgeZlibDeflateTune::body(
+    const Runtime::CallingFrame &Frame, uint32_t ZStreamPtr, int32_t GoodLength,
+    int32_t MaxLazy, int32_t NiceLength, int32_t MaxChain) {
+
+  const auto ZRes = SyncRun(Env, ZStreamPtr, Frame, [&](z_stream *HostZStream) {
+    return deflateTune(HostZStream, GoodLength, MaxLazy, NiceLength, MaxChain);
+  });
+
+  return ZRes;
+}
+
+Expect<int32_t>
+WasmEdgeZlibDeflateBound::body(const Runtime::CallingFrame &Frame,
+                               uint32_t ZStreamPtr, uint32_t SourceLen) {
+
+  const auto ZRes = SyncRun(Env, ZStreamPtr, Frame, [&](z_stream *HostZStream) {
+    return deflateBound(HostZStream, SourceLen);
+  });
+
+  return ZRes;
+}
+
+Expect<int32_t>
+WasmEdgeZlibDeflatePending::body(const Runtime::CallingFrame &Frame,
+                                 uint32_t ZStreamPtr, uint32_t PendingPtr,
+                                 uint32_t BitsPtr) {
+
+  auto *MemInst = Frame.getMemoryByIndex(0);
+  if (MemInst == nullptr) {
+    return Unexpect(ErrCode::Value::HostFuncError);
+  }
+
+  auto *Pending = MemInst->getPointer<uint32_t *>(PendingPtr);
+  auto *Bits = MemInst->getPointer<int32_t *>(BitsPtr);
+
+  const auto ZRes = SyncRun(Env, ZStreamPtr, Frame, [&](z_stream *HostZStream) {
+    return deflatePending(HostZStream, Pending, Bits);
+  });
+
+  return ZRes;
+}
+
+Expect<int32_t>
+WasmEdgeZlibDeflatePrime::body(const Runtime::CallingFrame &Frame,
+                               uint32_t ZStreamPtr, int32_t Bits,
+                               int32_t Value) {
+
+  const auto ZRes = SyncRun(Env, ZStreamPtr, Frame, [&](z_stream *HostZStream) {
+    return deflatePrime(HostZStream, Bits, Value);
+  });
+
+  return ZRes;
+}
+
+Expect<int32_t>
+WasmEdgeZlibDeflateSetHeader::body(const Runtime::CallingFrame &Frame,
+                                   uint32_t ZStreamPtr, uint32_t HeadPtr) {
+
+  auto HostGZHeader = std::make_unique<gz_header>();
+  auto HostGZHeaderPtr = HostGZHeader.get();
+
+  auto [It, _] = Env.GZHeaderMap.emplace(
+      std::pair<uint32_t, WasmEdgeZlibEnvironment::GZStore>{
+          ZStreamPtr, WasmEdgeZlibEnvironment::GZStore{
+                          .WasmGZHeaderOffset = HeadPtr,
```

Comment:
This is not a C++ standard. Don't use named fields.

Reply:
Removed it's usage.

Reply:
Fixed.